File: GeNeural/Neuron.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GeNeural {
    public class Neuron {
        private double[] weights;
        private double[] prevWeightDiff;
        public Neuron(params double[] weights) {
            this.weights = new double[weights.Length];
            for (int w = 0; w < weights.Length; w++) {
                this.weights[w] = weights[w];
            }
            prevWeightDiff = new double[weights.Length];
        }
        public double[] Weights {
            get { return weights; }
            private set {
                weights = value;
                prevWeightDiff = new double[weights.Length];

            }
        }
        public void AddWeights(params double[] weights) {
            double[] newWeights = new double[this.weights.Length + weights.Length];
            for (int i = 0; i < this.weights.Length; i++) {
                newWeights[i] = weights[i];
            }
            for (int i = 0; i < weights.Length; i++) {
                newWeights[this.weights.Length + i] = weights[i];
            }
            Weights = newWeights;
        }
        public void AddWeight(double weight = 0) {
            double[] newWeights = new double[weights.Length + 1];
            for (int i = 0; i < weights.Length; i++) {
                newWeights[i] = weights[i];
            }
            newWeights[newWeights.Length - 1] = weight;
            Weights = newWeights;
        }
        public void RemoveNeuronWeight(int neuronIndex) {
            int weightIndex = neuronIndex + 1;
            double[] newWeights = new double[weights.Length - 1];
            for (int w = 0; w < weightIndex; w++) {
                newWeights[w] = weights[w];
            }
            for (int w = weightIndex + 1; w < weights.Length; w++) {
                newWeights[w - 1] = weights[w];
            }
            Weights = newWeights;
        }
        public void SetWeights(double[] weights) {
            Weights = weights;
        }
        public double GetNeuronWeight(int neuronIndex) {
            return Weights[neuronIndex + 1];
        }
        public double GetThreshold() {
            return Weights[0];
        }
        public void SetWeight(int weightIndex, double weight) {
            Weights[weightIndex] = weight;
        }
        public void SetThresholdWeight(double weight) {
            Weights[0] = weight;
        }
        public void SetNeuronWeight(int neuronIndex, double weight) {
            Weights[neuronIndex + 1] = weight;
        }
        public double GetOutput(params double[] inputs) {
            double output = -weights[0];
            for (int i = 0; i < inputs.Length; i++) {
                output += weights[i + 1] * inputs[i];
            }
            return MathHelper.Sigmoid(output);
        }
    }
}

File: NeuralCLI/Neuron.cpp
#include "stdafx.h";
#include "Neuron.h";
#include "MathHelper.h"
namespace Neural {
	NeuronC::NeuronC(vector<double> weights) {
		this->weights = weights;
	}

	vector<double> NeuronC::GetWeights() {
		return this->weights;
	}

	void NeuronC::AddWeight(double weight = 0) {
		vector<double> newWeights(this->weights.size() + 1);
		for (int i = 0; i < this->weights.size(); i++) {
			newWeights[i] = this->weights[i];
		}
		newWeights[newWeights.size() - 1] = weight;
		this->SetWeights(newWeights);
	}

	void NeuronC::RemoveNeuronWeight(int neuronIndex) {
		int weightIndex = neuronIndex + 1;
		vector<double> newWeights(this->weights.size() - 1);
		for (int w = 0; w < weightIndex; w++) {
			newWeights[w] = this->weights[w];
		}
		for (int w = weightIndex + 1; w < this->weights.size(); w++) {
			newWeights[w - 1] = this->weights[w];
		}
		this->SetWeights(newWeights);
	}

	void NeuronC::SetWeights(vector<double> weights) {
		this->weights = weights;
	}

	double NeuronC::GetNeuronWeight(int neuronIndex) {
		return this->weights[neuronIndex + 1];
	}

	double NeuronC::GetThreshold() {
		return this->weights[0];
	}

	void NeuronC::SetWeight(int weightIndex, double weight) {
		this->weights[weightIndex] = weight;
	}

	void NeuronC::SetThresholdWeight(double weight) {
		this->weights[0] = weight;
	}

	void NeuronC::SetNeuronWeight(int neuronIndex, double weight) {
		this->weights[neuronIndex + 1] = weight;
	}

	double NeuronC::GetOutput(vector<double> inputs) {
		double output = -weights[0];
		for (int i = 0; i < inputs.size(); i++) {
			output += weights[i + 1] * inputs[i];
		}
		return MathHelper::Sigmoid(output);
	}
}